### PR TITLE
At/ihp40/request component detail

### PR DESCRIPTION
**Pull request by Angelbt12**
Se debe crear componente del detalle de la soliciud como lo muestra el [figma](https://www.figma.com/design/cpXEJp5TQdTGwVcnZ1YrIM/Employee-portal?node-id=184-4123&node-type=canvas&t=5uhvZpkRCWwKBD5U-0)

image

Criterios de aceptación

Debe tener las mismas medias que el mockup

Se debe presentar en el storybook

en esta rama se realizo unos ajustes a la modal para cumplir con lo requerido en la actividad ihp28

### CODE REVIEW

File: src/components/modals/ComponentDetailModal/styles.ts
Comment (novadeveloper14) [inline src/components/modals/ComponentDetailModal/styles.ts:1]: The component is too big on mobile to fit into mobile sizes which are 360px

image

You could follow the steps outlined in src\pages\holidays\modals\PendingHolidaysModal to comply with the measures.

File: src/components/modals/ComponentDetailModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+} from "./styles";
+
+export interface IModalContent {
+  label: string;
+  value: string;
+}
+
+export interface IRequestComponentDetailProps {
+  title: string;
+  handleClose: () => void;
```
Comment (novadeveloper14) [inline src/components/modals/ComponentDetailModal/index.tsx:27]: All functions must be ordered at the end

File: src/components/modals/ComponentDetailModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  buttonLabel: string;
+  modalContent: IModalContent[];
+  portalId?: string;
+}
+
+export const RequestComponentDetail = ({
+  title,
+  handleClose,
```
Comment (novadeveloper14) [inline src/components/modals/ComponentDetailModal/index.tsx:35]: All functions must be ordered at the end

File: src/components/modals/ComponentDetailModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+export interface IRequestComponentDetailProps {
+  title: string;
+  handleClose: () => void;
+  buttonLabel: string;
+  modalContent: IModalContent[];
+  portalId?: string;
+}
+
+export const RequestComponentDetail = ({
```
Comment (novadeveloper14) [inline src/components/modals/ComponentDetailModal/index.tsx:33]: change for a normal function and create props const

File: src/components/modals/ComponentDetailModal/stories/ComponentDetail.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import { useState } from "react";
+import { Meta, StoryObj } from "@storybook/react";
+import { Button } from "@inubekit/button";
+import { RequestComponentDetail, IRequestComponentDetailProps } from "../index";
```
Comment (novadeveloper14) [inline src/components/modals/ComponentDetailModal/stories/ComponentDetail.stories.tsx:4]: order imports

File: src/components/modals/ComponentDetailModal/styles.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
+import styled from "styled-components";
+import { inube } from "@inubekit/foundations";
+import { spacing } from "@design/tokens/spacing/spacing";
```
Comment (novadeveloper14) [inline src/components/modals/ComponentDetailModal/styles.ts:3]: order imports

**Review by novadeveloper14 — CHANGES_REQUESTED** [review]

File: src/components/modals/ComponentDetailModal/styles.ts (hunk 5 above)
Comment (Angelbt12) [reply src/components/modals/ComponentDetailModal/styles.ts:3]: ok

File: src/components/modals/ComponentDetailModal/stories/ComponentDetail.stories.tsx (hunk 4 above)
Comment (Angelbt12) [reply src/components/modals/ComponentDetailModal/stories/ComponentDetail.stories.tsx:4]: ok

File: src/components/modals/ComponentDetailModal/index.tsx
[the remark this answers is not in the document]
Comment (Angelbt12) [reply src/components/modals/ComponentDetailModal/index.tsx:1]: ok

**Review by novadeveloper14 — APPROVED** [review]

File: src/pages/certifications/components/CertificationsTable/index.tsx [hunk 6]
```diff
@@ -166,62 +169,96 @@ function CertificationsTable({
...
+
+  const modalContent = [
+    { label: "Número", value: "1234" },
+    { label: "Tipo", value: "Disfrute de vacaciones" },
+    { label: "Fecha", value: "22/Oct/2024" },
+    { label: "Estado", value: "En trámite de aprobación" },
+    { label: "Días de disfrute", value: "2" },
+    { label: "Destinatario", value: "A quien interese" },
+    { label: "Contrato", value: "Indefinido - 02/sep/2024" },
+  ];
```
Comment (diegoAndres01) [inline src/pages/certifications/components/CertificationsTable/index.tsx:189]: The fields and title are not being displayed as shown in figma
image
figma:
image

**Review by diegoAndres01 — CHANGES_REQUESTED** [review]

File: src/pages/certifications/components/CertificationsTable/index.tsx (hunk 6 above)
Comment (Angelbt12) [reply src/pages/certifications/components/CertificationsTable/index.tsx:189]: ok

File: src/components/modals/ComponentDetailModal/index.tsx [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+}
+
+import {
+  StyledContainerClose,
+  StyledContainerContent,
+  StyledModal,
+  StyledContainerTitle,
+} from "./styles";
+
+function RequestComponentDetail({
```
[the remark this answers is not in the document]
Comment (Angelbt12) [reply src/components/modals/ComponentDetailModal/index.tsx:32]: ok

**Review by diegoAndres01 — CHANGES_REQUESTED** [review]
In the mobile version the modal cannot be displayed

image

**Review by diegoAndres01 — APPROVED** [review]

**Pull request merged**